src/app/screens/Onboard/ConnectRaspiBlitz/index.tsx

```diff
@@ -16,10 +16,17 @@ export default function ConnectRaspiBlitz() {
   const [loading, setLoading] = useState(false);
 
   function handleUrl(event: React.ChangeEvent<HTMLInputElement>) {
-    setFormData({
-      ...formData,
-      [event.target.name]: `https://${event.target.value.trim()}`,
-    });
+    if (event.target.value.substring(0, 4) === "http") {
```

Comment:
I'd do something like this. It is not good to duplicate the function calls here, just for adding the `https://` - just put it in a variable and update that if needed. 

Everytime you duplicate some code ask if that is really needed or if it can somehow be refactored. 
Duplication easily leads to errors. e.g. when you want to update this code here and miss one of the setFormData calls... 

```js
let url = event.target.value.trim();
if (event.target.value.substring(0, 4) !== "http") {
  url = `https://${url}`;
}
...
```
this should work, doesn't it?

Reply:
Yep I just tested it and it works, I wasn't sure if it would but it does. I didn't think to create a variable in this way to DRY this function. Thanks for the suggestion!